ytt/001-schemas/README.md

```diff
@@ -239,111 +900,393 @@ Beyond specifying a type for a value, one can specify more dynamic constraints o
+When schema is present, Data Values documents imply an automatic merge of new keys. This is 
+equivalent to all `@data/values` documents being annotated with `@overlay/match-child-defaults missing_ok=True`.
 
-#### `@schema/doc`
+### Overlays of Schema
 
-  This annotation is a way to add a longer description to the section of a schema associated with a key, similar to the JSON Schema description field.
+TBD
 
-  ```yaml
-  #@schema/doc "The user password used to log in to the system"
-  user_password: ""
-  ```
+- It must be possible to extend Schema (i.e. add new items)
+- Is it necessary to be able to modify Schema?
 
-#### `@schema/example`
 
-  This annotation will take one argument that is an example of a value that satisfies the schema.
+## Open Questions
 
-  ```yaml
-  #@schema/example "my_domain.example.com"
-  system_domain: ""
-  ```
+- It is not uncommon for a validation to want to refer to a sibling node.
+    - right now, the Configuration Author places these on the parent node.
+    - that puts separation between the node being constrained and its constraint.
+    - is there a feasible way for a validation rule to get access to its siblings?
+- Documentation could be enhanced if validation rules were included. How could validation authors
+  provide documentation for their validation?
+  - provide a magic string for the input and the function returns documentation?
+  - function always returns documentation as part of return value?
+- How should schema work with ytt libraries?
+  - How do we support data values that are targetting a dependency library?
+- Are there needs around versioning schema?
+- How will code defined in a schema file make its way into the execution context of the target "template"?
+  - e.g. function supporting validation
+  - idea: could we capture the "globals" resulting from the execution of the schema "template" and feed those into the target "template"'s execution context?
+  - there are expressions/statements to consume during the schema "template" construction... but other expression/statements that need to be made available to target "templates".
+- Should we support generating `ytt` Schema from other popular sources (e.g. JSON Schema, OpenAPI v3 schema)?
+- how to add more things to schema? (via overlays similar to data values)
+- provide programmatic schema.apply() (similar to overlay.apply)
+- respect schema for data values set via cmd line flags/env vars
+- add --data-values-schema-inspect (similar to --data-values-inspect)
+  - generate html view via builtin server
+- with the addition of a schema, the first data values file no longer serves that role. Could we improve the UX of adding multiple data value files by not requiring subsequent data values to be overlays?
+- could a user include both a data/value + schema/amend in the same file?
+  - allowing a user to define and include input values in one file?
```

Comment:
Right now, no.

- Schema must be prepared before Data Values can be processed... which in turn must be prepared before templates can be executed; this is a hard constraint.
- We currently construct a single execution context for each file passed in. If a file contains two logical execution contexts (e.g. one for a Data Values document — which may depend on variables or functions defined in _other_ documents and another for a template that depends on the `data` module), we currently do not have a way to separate them.
  - perhaps the document end literal `...` could be used to denote an execution context boundary, here?